node_package/src/react_on_rails.js

```diff
@@ -0,0 +1,277 @@
+    if (React.version >= '0.14') {
+      console.warn(
+        'WARNING: `react-dom/server` is not configured in webpack.server.rails.config.js file as an entry.\n' +
+        'See: https://github.com/shakacode/react_on_rails/blob/master/docs/webpack.md for more detailed hints.');
+    }
+
+    return context.React;
+  }
+
+  return context.ReactDOMServer;
+}
+
+function wrapInScriptTags(scriptBody) {
+  if (!scriptBody) {
+    return '';
+  }
+
+  return '\n<script>' + scriptBody + '\n</script>';
+}
+
+function forEachComponent(fn) {
+  const els = document.getElementsByClassName('js-react-on-rails-component');
+  for (let i = 0; i < els.length; i++) {
+    fn(els[i]);
+  }
+}
+
+function unmount(el) {
+  const domId = el.getAttribute('data-dom-id');
+  const domNode = document.getElementById(domId);
+  provideClientReact().unmountComponentAtNode(domNode);
+}
+
+function isRouterResult(reactElementOrRouterResult) {
+  return !!(reactElementOrRouterResult.redirectLocation ||
+  reactElementOrRouterResult.error);
+}
+
+function createReactElementOrRouterResult(componentName, props, domId, trace, generatorFunction, location) {
+  if (trace) {
+    console.log('RENDERED ' + componentName + ' to dom node with id: ' + domId);
+  }
+
+  const component = ReactOnRails.componentForName(componentName);
+  if (generatorFunction) {
+    return component(props, location);
+  }
+
+  return React.createElement(component, props);
+}
+
+function render(el) {
+  const componentName = el.getAttribute('data-component-name');
+  const domId = el.getAttribute('data-dom-id');
+  const props = JSON.parse(el.getAttribute('data-props'));
+  const trace = JSON.parse(el.getAttribute('data-trace'));
+  const generatorFunction = JSON.parse(el.getAttribute('data-generator-function'));
+  const expectTurboLinks = JSON.parse(el.getAttribute('data-expect-turbo-links'));
+
+  if (!turbolinksInstalled && expectTurboLinks) {
+    console.warn('WARNING: NO TurboLinks detected in JS, but it is in your Gemfile');
+  }
+
+  try {
+    const domNode = document.getElementById(domId);
+    if (domNode) {
+      const reactElementOrRouterResult = createReactElementOrRouterResult(componentName, props,
+        domId, trace, generatorFunction);
+      if (isRouterResult(reactElementOrRouterResult)) {
+        throw new Error('You returned a server side type of react-router error: ' +
+          JSON.stringify(reactElementOrRouterResult) +
+          '\nYou should return a React.Component always for the client side entry point.');
+      } else {
+        provideClientReact().render(reactElementOrRouterResult, domNode);
+      }
+    }
+  } catch (e) {
+    ReactOnRails.handleError({
+      e: e,
+      componentName: componentName,
+      serverSide: false,
+    });
+  }
+}
+
+function reactOnRailsPageLoaded() {
+  forEachComponent(render);
+}
+
+function reactOnRailsPageUnloaded() {
+  forEachComponent(unmount);
+}
+
+const components = {};
+
+ReactOnRails = {
```

Comment:
Should we do something like `export default ReactOnRails;` for other file's `import`? Hmm, but it doesn't work when I added it.